presto-main/src/main/java/com/facebook/presto/server/CatalogServerModule.java

```diff
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.airlift.configuration.AbstractConfigurationAwareModule;
+import com.facebook.airlift.discovery.server.EmbeddedDiscoveryModule;
+import com.facebook.presto.catalogserver.CatalogServer;
+import com.facebook.presto.dispatcher.NoOpQueryManager;
+import com.facebook.presto.execution.QueryManager;
+import com.facebook.presto.execution.resourceGroups.NoOpResourceGroupManager;
+import com.facebook.presto.execution.resourceGroups.ResourceGroupManager;
+import com.facebook.presto.failureDetector.FailureDetectorModule;
+import com.facebook.presto.metadata.CatalogManager;
+import com.facebook.presto.transaction.ForTransactionManager;
+import com.facebook.presto.transaction.InMemoryTransactionManager;
+import com.facebook.presto.transaction.TransactionManager;
+import com.facebook.presto.transaction.TransactionManagerConfig;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+
+import javax.inject.Singleton;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.facebook.airlift.configuration.ConditionalModule.installModuleIf;
+import static com.facebook.airlift.discovery.client.DiscoveryBinder.discoveryBinder;
+import static com.facebook.airlift.http.client.HttpClientBinder.httpClientBinder;
+import static com.facebook.airlift.jaxrs.JaxrsBinder.jaxrsBinder;
+import static com.facebook.drift.server.guice.DriftServerBinder.driftServerBinder;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+
+public class CatalogServerModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        // discovery server
+        install(installModuleIf(EmbeddedDiscoveryConfig.class, EmbeddedDiscoveryConfig::isEnabled, new EmbeddedDiscoveryModule()));
+
+        // presto coordinator announcement
+        discoveryBinder(binder).bindHttpAnnouncement("presto-catalog-server");
+
+        // failure detector
+        binder.install(new FailureDetectorModule());
+        jaxrsBinder(binder).bind(NodeResource.class);
+        jaxrsBinder(binder).bind(WorkerResource.class);
+        httpClientBinder(binder).bindHttpClient("workerInfo", ForWorkerInfo.class);
+
+        // query manager
+        binder.bind(QueryManager.class).to(NoOpQueryManager.class).in(Scopes.SINGLETON);
+
+        // resource group manager
+        binder.bind(ResourceGroupManager.class).to(NoOpResourceGroupManager.class);
```

Comment:
Why is this needed?

Reply:
Server requires that an resourceGroupManager is assigned upon startup or it complains. Thus it is assigned to the NoOp version which is basically the same thing as not having it at all